README.md

```diff
@@ -1,6 +1,89 @@
-You need to implement the following tasks:
-1. Write a method to iterate through all elements in a HashSet containing the days of the week (strings). The method should return a string containing the elements of the set, separated by a space.
-2. Write a method to clone a HashSet to another HashSet. The method should return a string containing the elements of the set, separated by a space.
-3. Write a method to compare two sets and retain elements that are the same on both sets.First HashSet should contain "Red", "Green","Black", "White" elements, second HashSet should contain "Red", "Pink", "Black", "Orange" elements.  The method should return a string containing the elements of the first set, separated by a space.
-4. Write a method to create a new TreeSet, add all the days of the week (string). The method should return a string containing the elements of the set, separated by a space.
-5. Write a method to find the numbers less than 10 in a new TreeSet. The method should return a string containing the elements of the set, separated by a space.
+# Tasks
+
+## 1. Create HashSet
+
+Implement the program which gets `List<Integer>` as parameter and creates a `HashSet<Integer>` following the rule:
+
+* If the element `x` from `List<Integer>` is even, then add to the `HashSet` values: `x`, `x/2`, `(x/2)/2`
+  ... `((x/2)/2)../2)` (dividing by 2 until the last inserted element is an odd number)
+* If the element `x` from `List<Integer>` is odd, then add to the `HashSet`values:`x` and `2x`
+
+```
+public class HashSetCreator {
+    public HashSet<Integer> createHashSet(List<Integer> sourceList) {
+    }
+}
+```
+
+### Example
+
+**Input**
+
+```
+sourceList = [2, -1, 3, 8, -5, 12, 10]
+```
+
+**Output**
+
+```
+[2, 1, -1, -2, 3, 6, 8, 4, -5, -10, 12]
+```
+
+## 2. Create Set of squares
+
+Implement the program which creates `TreeSet<Integer>` of squares of elements from `List<Integer>` and returns a `subset
+of this set`, consisting of elements in the range `[lowerBound, upperBound)`. To get subset use method
```

Comment:
As I can see,  `lowerBound <= x < upperBound`. It's right? Or do you need use the same brackets?

Reply:
Yes, it's right

Reply:
It's a good pount to use  `lowerBound<= x <= upperBound`, so in this case we need to configure ` subSet(E fromElement, boolean fromInclusive, E toElement,   boolean toInclusive) function.`